src/export/StructureDefinitionExporter.ts

```diff
@@ -178,6 +175,107 @@ export class StructureDefinitionExporter implements Fishable {
     }
   }
 
+  /**
+   * Handles a ContainsRule that is on an extension path, appropriately exporting it as a reference to a standalone
+   * extension or an inline extension.
+   * @param {Profile|Extension} fshDefinition - the FSH Definition the rule is on
+   * @param {ContainsRule} rule - the ContainsRule that is on an extension element
+   * @param {StructureDefinition} structDef - the StructDef of the resulting profile or element
+   * @param {ElementDefinition} element - the element to apply the rule to
+   */
+  private handleExtensionContainsRule(
+    fshDefinition: Profile | Extension,
+    rule: ContainsRule,
+    structDef: StructureDefinition,
+    element: ElementDefinition
+  ) {
+    if (!element.slicing) {
+      element.sliceIt('value', 'url');
+    }
+    rule.items.forEach(item => {
+      if (item.type) {
+        const extension = this.fishForMetadata(item.type, Type.Extension);
+        if (extension == null) {
+          logger.error(
+            `Cannot create ${item.name} extension; unable to locate extension definition for: ${item.type}.`,
+            rule.sourceInfo
+          );
+          return;
+        }
+        const slice = element.addSlice(item.name);
+        if (!slice.type[0].profile) {
+          slice.type[0].profile = [];
+        }
+        slice.type[0].profile.push(extension.url);
+      } else {
+        const external = this.fishForMetadata(item.name, Type.Extension);
+        if (external) {
+          // Log a warning. This is intended to help users transitioning from SUSHI 0.9.x to SUSHI 0.10.0,
+          // since the syntax for specifying external extensions has changed.
+          // First find the corresponding card rule to assist in constructing the warning message
+          let card = 'm..n';
+          const cardRule = fshDefinition.rules.find(
+            r => r instanceof CardRule && r.path === `${rule.path}[${item.name}]`
+          );
+          if (cardRule) {
+            card = `${(cardRule as CardRule).min}..${(cardRule as CardRule).max}`;
+          }
+          // Customize the message based on whether or not extension comes from the tank
+          const fshMeta = this.tank.fishForMetadata(item.name);
+          if (fshMeta && fshMeta.url === external.url) {
```

Comment:
I'm a little confused by the `fshMeta.url === external.url` check. What is this check necessary for?

Reply:
*In theory*, if `fshMeta` is not null, then the URLs should probably always match.  BUT... that assumes that when `item.name` was resolved, that the tank was searched _first_, before core FHIR or any external IGs.  I put the URL check in as insurance in case the tank is not searched first (particularly if any code in that area changes).  The URL is the only guaranteed unique identifier (as opposed to `name`, which can have multiple matches).

It's probably unnecessary 99.99% of the time, but... it's also harmless (as far as I can tell).

Reply:
Ok that's fair, if we change our resolution order then I could see this being meaningful, and you're right I can't see how it would hurt. So I'm alright with it!